ui/src/app/job-list/job-list.component.ts

```diff
@@ -151,6 +173,26 @@ export class JobListComponent implements OnInit {
     this.jobStream.loadAtLeast((e.pageIndex+2) * e.pageSize)
       .catch((error) => this.handleError(error));
   }
+
+  toggleDisplayColumn(field: DisplayField) {
+    const newValue = !field.primary;
+    this.displayFields.forEach((df) => {
+      if (df.field == field.field) {
+        df.primary = newValue;
+      }
+    })
+  }
+
+  saveSettings() {
+    let fields: string[] = [];
+    this.displayFields.forEach((field) => {
+      if (field.primary) {
+        fields.push(field.field);
+      }
+    })
+    this.settingsService.setDisplayColumns(fields, this.projectId);
+    this.jobTable.setUpFieldsAndColumns();
```

Comment:
Do you really need to call into the child component here? Can't you just update `this.displayFields`, thereby updating the child component which binds that value?